### PR TITLE
Gravity Bridge base

### PR DESCRIPTION
**Pull request by ToasterTheBrave**
# Related Github tickets

- https://github.com/VolumeFi/paloma/issues/431

# Background

This adds the base functionality previously developed for the cosmos gravity bridge.  It has been updated to work with the new version of cosmos-sdk.

# Testing completed

- [x] test coverage exists or has been added/updated
- [x] tested in a private testnet

# Breaking changes

- [x] I have checked my code for breaking changes
- [x] If there are breaking changes, there is a supporting migration.

### CODE REVIEW

**Review by taariq — APPROVED** [review]
A great start. Let's get to testing.

File: proto/palomachain/paloma/gravity/genesis.proto [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+// interfaces
+message GenesisState {
+  Params params = 1;
+  uint64 last_observed_event_nonce = 2;
+  repeated google.protobuf.Any outgoing_txs = 3;
+  repeated google.protobuf.Any confirmations = 4;
+  repeated EthereumEventVoteRecord ethereum_event_vote_records = 9;
+  repeated MsgDelegateKeys delegate_keys = 10;
+  repeated ERC20ToDenom erc20_to_denoms = 11;
+  repeated SendToEthereum unbatched_send_to_ethereum_txs = 12;
```
Comment (byte-bandit) [inline proto/palomachain/paloma/gravity/genesis.proto:116]: I suggest adding `(gogoproto.nullable) = false` to all of these, unless there's an actual difference here between empty set and a nil set.

Comment (byte-bandit) [reply proto/palomachain/paloma/gravity/genesis.proto:116]: Same for other cases of `repeated` fields in the changed proto files.

**Comment by taariq** [comment]
@ToasterTheBrave is this okay to merge and prep for release?

**Comment by ToasterTheBrave** [comment]
@taariq I am going to quickly address @byte-bandit's request before we merge

**Comment by ToasterTheBrave** [comment]
@byte-bandit As I'm getting into this, It's quite a bit of a refactor.  I would like to punt this to later so that we can get this included in today's release.  It's an optimization, not a bug necessarily.  Are you fine with that?

**Pull request merged**